lib/pkg_editor/src/zlib.c

```diff
@@ -116,6 +116,11 @@ static void zlib_load() {
 #ifdef _MSC_VER
 #pragma warning(pop)
 #endif
+#ifdef _WIN32
+  FreeLibrary(library);
```

Comment:
I am not sure if this caused the segfault, but I think it should be `FreeLibrary((HMODULE)library);`, `zlib_load_library` returned a `void *` but `FreeLibrary` expected a `HMODULE`

Reply:
I believe what is happening is that calling `dlclose` assumes that we are done with the library and so frees the memory, but does not replace that area in memory with invalid data. So, calling the functions from the DLL still work, although it's not being used as intended. (https://pubs.opengroup.org/onlinepubs/7908799/xsh/dlclose.html)

FreeLibrary probably does invalidate the data where the pointer pointed to when called, although I could not find any documentation explicitly mentioning this.